app/models/journey.rb

```diff
@@ -43,8 +45,26 @@ class Journey < ApplicationRecord
 
   after_initialize :initialize_state # NB there is an equivalent after(:build) callback used by FactoryBot in the journeys factory
 
+  def vehicle
+    get_metadata('vehicle')
```

Comment:
meta data seems like a strange term to use here - I would have thought that the vehicle used for the journey is more concrete than that, and makes sense to have as an explicit attribute on the model?

Reply:
@smoothcontract actually, it is not really that well defined at all... I did consider a more concrete implementation _a la_ the `Types::JSONB` approach. But I'm wary about being over prescriptive on the format of the vehicle record without more information from the suppliers. Allowing this to be any json seemed pragmatic for now.

Reply:
actually, reflecting on this a bit more, I think you are right that vehicle should be an explicit attribute. At the time I envisaged lots of other metadata about the journey, so having one place to dump it all seemed sensible. But for now its probably just vehicle, so I'll change this 👍

Reply:
If the vehicle attributes might change over time then using a JSONB column might be a good choice - and gives the ability to wrap it in a typed class in future once we better understand the data structure. Having a dedicated `vehicle` attribute for this rather than merging it under `details` feels like a good design choice 🤞